pkg/util/tracingservice/tests/service_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/cockroach/pkg/util/tracingservice"
+	"github.com/cockroachdb/cockroach/pkg/util/tracingservice/tracingservicepb"
+	"github.com/gogo/protobuf/types"
+	"github.com/stretchr/testify/require"
+)
+
+// testStructuredImpl is a testing implementation of Structured event.
+type testStructuredImpl struct {
+	*types.StringValue
+}
+
+var _ tracing.Structured = &testStructuredImpl{}
+
+func (t *testStructuredImpl) String() string {
+	return fmt.Sprintf("structured=%s", t.Value)
+}
+
+func newTestStructured(i string) *testStructuredImpl {
+	return &testStructuredImpl{
+		&types.StringValue{Value: i},
+	}
+}
+
+func setupTraces(t1, t2 *tracing.Tracer) (uint64, uint64, func()) {
```

Comment:
I'd benefit from a top-level comment here, perhaps with a diagram, that explains the trace structure being setup here. It'd then be easy to answer at a quick glance what the trace should look like from each observer (t1, t2, for each traceID/span).

Reply:
Done, hope the diagram is along the lines of what you were thinking.